packages/poseidon/test/utils.ts

```diff
@@ -0,0 +1,20 @@
+import { Project } from '@zk-kit/artifacts'
+import type { PoseidonProof } from '@zk-kit/poseidon-proof'
+import { unpackGroth16Proof } from '@zk-kit/utils'
+import { keccak256, toBeHex } from 'ethers'
+import { readFileSync } from 'node:fs'
+import { join } from 'node:path'
+import { groth16 } from 'snarkjs'
+
+export const SCOPE = 'scope'
+
+const hash = (message: string) => (BigInt(keccak256(toBeHex(message, 32))) >> BigInt(8)).toString()
+
+export const verifyPoseidonProof = (
+  { digest, numberOfInputs, proof, scope }: PoseidonProof,
+) => {
+  const verifKey = JSON.parse(
+    readFileSync(join(__dirname, '..', `${Project.POSEIDON}-${numberOfInputs}.json`), 'utf8'),
+  )
+  return groth16.verify(verifKey, [digest, hash(scope)], unpackGroth16Proof(proof))
+}
```

Comment:
instead of having to define this util in order to be able to use a specific verification key file, we should consider adding an optional arg to allow overriding the verification key file used for verification. 
https://github.com/privacy-scaling-explorations/zk-kit/blob/2173bc800d2a22409678a7d3983df48d4f8efd18/packages/poseidon-proof/src/verify.ts#L12

Like we do for [`generate`](https://github.com/privacy-scaling-explorations/zk-kit/blob/2173bc800d2a22409678a7d3983df48d4f8efd18/packages/poseidon-proof/src/generate.ts#L28) functions,

Reply:
Isn't this file the same as `packages/test/index.ts`?

Reply:
that;s really weird, I don't have these files in my git environment. I am probably in some weird sparse checkout state